### PR TITLE
Flag for disabling autoscale

### LINKED ISSUE #435: Add disable flag to the 'autoscale' section
@rigazilla as idea, maybe we can also add enable flag to the  _autoscale_ section as it contains several options. When user will try to playground with this options, it would be nice to easy enable/disable without add/remove spec section.

_Originally posted by @dmvolod in https://github.com/infinispan/infinispan-operator/pull/429_

### PR DESCRIPTION
**Pull request by rigazilla**
This resolves #435 
a bool field `disabled` has been added to the `autoscale` struct

### CODE REVIEW

File: pkg/controller/infinispan/infinispan_controller.go [hunk 1]
```diff
@@ -410,7 +410,7 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 
 	// All pods ready start autoscaler if needed
-	if infinispan.Spec.Autoscale != nil {
+	if infinispan.Spec.Autoscale != nil && infinispan.Spec.Service.Type == infinispanv1.ServiceTypeCache {
```
Comment (dmvolod) [inline pkg/controller/infinispan/infinispan_controller.go:413]: We have an IsDataGrid() for easy validation

File: deploy/operator-install.yaml [hunk 2]
```diff
@@ -174,6 +174,8 @@ spec:
             autoscale:
               description: Autoscale describe autoscaling configuration for the cluster
               properties:
+                disabled:
```
Comment (dmvolod) [inline deploy/operator-install.yaml:177]: Looks like code-gen need to be run again

Comment (rigazilla) [reply deploy/operator-install.yaml:177]: what's missing here? `make codegen` shouldn't be enough?

Comment (dmvolod) [reply deploy/operator-install.yaml:177]: I'm not sure, but some comments can be missed

Comment (dmvolod) [reply deploy/operator-install.yaml:177]: Yes, please run `make codegen` again

Comment (rigazilla) [reply deploy/operator-install.yaml:177]: `make codegen` has nothing more to add. seems we're ok

**Comment by Crumby** [comment]
@rigazilla I'm probably late on this one but this design doesn't fit k8s in my opinion. Feature being `disabled` is simply done by not-being defined. `disabled` field is in this case redundant and only present for lazy users. Also if this feature has disable field, shouldn't other features have one eg. xsite?

Place where `disabled` would make sense are features enabled by default, eg.: encryption, authentication.

**Pull request merged**